src/raven/integrations/grip/exception_handler.cr

```diff
@@ -0,0 +1,71 @@
+require "http"
+require "../http/*"
+
+module Raven
+  module Grip
+    # Exception handler capturing all unhandled `Exception`s.
+    # After capturing exception is re-raised.
+    #
+    # ```
+    # class Application < Grip::Application
+    #   def initialize
+    #     super(environment: "development")
+    #
+    #     # By default the router has 4 entries, you need to insert
+    #     # a handler before the exception handler.
+    #     #
+    #     # [
+    #     #   exception_handler,
+    #     #   pipeline_handler,
+    #     #   websocket_handler,
+    #     #   http_handler,
+    #     # ]
+    #     router.insert(0, Raven::Grip::ExceptionHandler.new(self))
+    #   end
+    # end
+    # ```
+    class ExceptionHandler
+      include HTTP::Handler
+      include Raven::HTTPHandler
+
+      # See `::Grip::Routers::Route`
+      CULPRIT_PATTERN_KEYS = %i(method path)
+
+      def initialize(
+        @application : ::Grip::Application,
+        @culprit_pattern = "%{method} %{path}",
+        @capture_data_for_methods = %w(POST PUT PATCH),
+        @default_logger = "grip",
+      )
+      end
+
+      private def route_from_context(context : HTTP::Server::Context)
+        router = @application.http_handler
+
+        route = router.find_route(context.request.method, context.request.path)
+        route = router.find_route("ALL", context.request.path) unless route.found?
+
+        route if route.route_found?
+      end
```

Comment:
@grkek Ideally, I wouldn't need to duplicate this, although I couldn't find a cleaner way to associate the context with the matching route.